bsp/m5stack_core/include/bsp/m5stack_core.h

```diff
@@ -0,0 +1,387 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ESP BSP: M5Stack Core
+ */
+
+#pragma once
+
+#include "sdkconfig.h"
+#include "driver/gpio.h"
+#include "driver/i2c.h"
+#include "driver/sdmmc_host.h"
+#include "esp_codec_dev.h"
+#include "bsp/config.h"
+#include "bsp/display.h"
+
+#include "driver/i2s_std.h"
+
+#if (BSP_CONFIG_NO_GRAPHIC_LIB == 0)
+#include "lvgl.h"
+#include "esp_lvgl_port.h"
+#endif // BSP_CONFIG_NO_GRAPHIC_LIB == 0
+
+/**************************************************************************************************
+ *  BSP Capabilities
+ **************************************************************************************************/
+
+#define BSP_CAPS_DISPLAY        1
+#define BSP_CAPS_TOUCH          0
+#define BSP_CAPS_BUTTONS        1
+#define BSP_CAPS_AUDIO          1
+#define BSP_CAPS_AUDIO_SPEAKER  1
+#define BSP_CAPS_AUDIO_MIC      0
+#define BSP_CAPS_SDCARD         1
+#define BSP_CAPS_IMU            0
+
+/**************************************************************************************************
+ *  M5Stack-Core-2 pinout
+ **************************************************************************************************/
+/* I2C */
+#define BSP_I2C_SCL           (GPIO_NUM_22)
+#define BSP_I2C_SDA           (GPIO_NUM_21)
+/* Audio */
+#define BSP_I2S_SCLK          (GPIO_NUM_12)
+#define BSP_I2S_MCLK          (GPIO_NUM_0)
+#define BSP_I2S_LCLK          (GPIO_NUM_13)
+#define BSP_I2S_DOUT          (GPIO_NUM_15)
+#define BSP_I2S_DSIN          (GPIO_NUM_34)
+#define BSP_SPEAKER_IO        (GPIO_NUM_25)
+#define BSP_POWER_AMP_IO      (GPIO_NUM_NC)
+#define BSP_MUTE_STATUS       (GPIO_NUM_NC)
+/* Display */
+#define BSP_LCD_MOSI          (GPIO_NUM_23)
+#define BSP_LCD_MISO          (GPIO_NUM_NC)
+#define BSP_LCD_PCLK          (GPIO_NUM_18)
+#define BSP_LCD_CS            (GPIO_NUM_14)
+#define BSP_LCD_DC            (GPIO_NUM_27)
+#define BSP_LCD_RST           (GPIO_NUM_33)
+#define BSP_LCD_BACKLIGHT     (GPIO_NUM_32)
+#define BSP_LCD_TOUCH_INT     (GPIO_NUM_NC)
+
+/* SD card */
+#define BSP_SD_MOSI           (GPIO_NUM_23)
+#define BSP_SD_MISO           (GPIO_NUM_19)
+#define BSP_SD_SCK            (GPIO_NUM_18)
+#define BSP_SD_CS             (GPIO_NUM_4)
+
+/* BUTTON */
+#define BSP_BUTTON_A          (GPIO_NUM_39)
+#define BSP_BUTTON_B          (GPIO_NUM_38)
+#define BSP_BUTTON_C          (GPIO_NUM_37)
```

Comment:
What about set buttons same like here: https://github.com/espressif/esp-bsp/blob/9ba025e1361f3252dc2be8c1aafcf69746544fd1/bsp/esp-box-lite/esp-box-lite.c#L375
Like a navigation buttons for LVGL in default.